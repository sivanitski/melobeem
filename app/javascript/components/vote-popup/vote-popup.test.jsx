import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import VotePopup from "./vote-popup";

describe("Components", () => {
  describe("VotePopup", () => {
    it("should render VotePopup component", () => {
      const component = renderer
        .create(<VotePopup handlePopupClose={() => {}} />)
        .toJSON();

      expect(component).toMatchSnapshot();
    });

    it("should call onClick method", () => {
      const mockCallBack = jest.fn();
      const component = shallow(<VotePopup handlePopupClose={mockCallBack} />);
      expect(mockCallBack.mock.calls.length).toBe(0);
      component.find(".popup__close").simulate("click");
      expect(mockCallBack.mock.calls.length).toBe(1);
    });
  });
});
