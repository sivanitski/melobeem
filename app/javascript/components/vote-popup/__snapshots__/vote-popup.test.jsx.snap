// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components VotePopup should render VotePopup component 1`] = `
<div
  className="popup"
>
  <div
    className="popup__inner"
  >
    <div
      className="popup__close"
      onClick={[Function]}
    >
      <test-file-stub />
    </div>
    <div
      className="popup__avatar"
    >
      <img
        src="test-file-stub"
      />
    </div>
    <div
      className="popup__share headline"
    >
      Share
    </div>
    <div
      className="popup__text text-grey"
    >
      Every Facebook post 
      <br />
       may make Diana closer to 
      <br />
       the wining point
    </div>
    <div
      className="popup__button"
    >
      Share on Facebook
    </div>
  </div>
</div>
`;
